src/server/terrain/simple_structures/SimpleTreeModel.zig

```diff
@@ -57,37 +56,58 @@ pub fn loadModel(arenaAllocator: NeverFailingAllocator, parameters: ZonElement)
 	return self;
 }
 
+fn getWoodBlock(self: *SimpleTreeModel, data: u16) main.blocks.Block {
+	if(self.woodBlock.mode() == main.rotation.getByID("log")) {
+		return .{.typ = self.woodBlock.typ, .data = data};
+	}
+	return self.woodBlock;
+}
+
 pub fn generateStem(self: *SimpleTreeModel, x: i32, y: i32, z: i32, height: i32, chunk: *main.chunk.ServerChunk, seed: *u64) void {
 	if(chunk.super.pos.voxelSize <= 2) {
 		var pz: i32 = chunk.startIndex(z);
 		while(pz < z + height) : (pz += chunk.super.pos.voxelSize) {
 			if(chunk.liesInChunk(x, y, pz)) {
-				chunk.updateBlockIfDegradable(x, y, pz, if(pz == z + height - 1) self.topWoodBlock else self.woodBlock);
+				var data: u16 = 0;
+				if(pz != z + height - 1) {
+					data |= @as(u16, 1) << @intFromEnum(Neighbor.dirUp);
+				}
+				data |= @as(u16, 1) << @intFromEnum(Neighbor.dirDown);
 
 				if(self.branched) {
 					const chance = @sqrt(@as(f32, @floatFromInt(pz - z))/@as(f32, @floatFromInt(height*2)));
 					if(main.random.nextFloat(seed) < chance) {
 						const d = main.random.nextIntBounded(u32, seed, 4);
-						generateBranch(self, x, y, pz, d, chunk, seed);
+						if(generateBranch(self, x, y, pz, d, chunk, seed)) |side| {
+							data |= @as(u16, 1) << @intFromEnum(side);
+						}
 					}
 				}
+
+				chunk.updateBlockIfDegradable(x, y, pz, self.getWoodBlock(data));
 			}
 		}
 	}
 }
 
-pub fn generateBranch(self: *SimpleTreeModel, x: i32, y: i32, z: i32, d: u32, chunk: *main.chunk.ServerChunk, seed: *u64) void {
+pub fn generateBranch(self: *SimpleTreeModel, x: i32, y: i32, z: i32, d: u32, chunk: *main.chunk.ServerChunk, seed: *u64) ?Neighbor {
 	_ = seed;
 
 	if(d == 0 and chunk.liesInChunk(x + 1, y, z)) {
-		chunk.updateBlockIfDegradable(x + 1, y, z, .{.typ = self.topWoodBlock.typ, .data = 2});
+		chunk.updateBlockIfDegradable(x + 1, y, z, self.getWoodBlock(@as(u16, 1) << @intFromEnum(Neighbor.dirNegX)));
```

Comment:
There is a function for this
```suggestion
		chunk.updateBlockIfDegradable(x + 1, y, z, self.getWoodBlock(Neighbor.dirNegX.bitMask()));
```